doc/proposals/scorecard-custom-tests.md

```diff
@@ -0,0 +1,403 @@
+ * Scorecard Displays Test Results
+
+#### Phase 2
+
+For consideration, with the above custom test capability, there is benefit to Red Hat in migrating the existing scorecard internal tests to be based on the custom test format.  For this document, I’ve called this work, Phase 2, and is optional but worth consideration.  
+
+Today there are 2 internal scorecard tests that make use of the scorecard proxy, with these tests migrating to the custom test format, the scorecard proxy is likely not necessary which would also reduce the overall scorecard implementation complexity.
+
+#### Building the Proposed Design
+
+The proposed design was influenced greatly by the KUDO testing framework in various concepts, however, the proposed design doesn’t necessarily have to include KUDO as a dependency.  A decision of whether or not to incorporate KUDO testing APIs within the Red Hat solution needs to be reached.  Some pro’s and con’s to using KUDO testing frameworks are listed below.  If KUDO’s testing framework is not incorporated, then a custom built set of equivalent capabilities would need to be developed.
+The advantage of using Kuttl as the test framework in scorecard would include:
+
+ * Pre-built, works, community, evolving.
+ * Ready to work with, code exists and mostly works, a starting point exists.
+ * Handles the basic test case, that being to test Kube resource state
+ * Written in golang, making integration into scorecard a possibility.
+
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+### Upgrade / Downgrade Strategy
+
+To implement this custom test functionality, additions or changes to the existing scorecard configuration file and command line flags would be necessary and would not be supported in previous scorecard releases.
+
+## Implementation History
+
+ * Proposal initiated in March 2020.
+
+## Drawbacks
+
+Drawbacks of using Kuttl within this proposed design might include:
+
+ * Immaturity of Kuttl and KUDO in general, integration into Operator SDK will require Redhat to possibly contribute upstream to the Kuttl project itself in order to meet the needs of the Operator SDK.
+  * <https://github.com/kudobuilder/kuttl/issues/16#issuecomment-589242819>
+  * <https://github.com/kudobuilder/kuttl/issues/12>
+  * <https://github.com/kudobuilder/kudo/blob/master/keps/0008-operator-testing.md>
+  * <https://github.com/kudobuilder/kuttl/pull/5>
+
+ * Dependency on a community project for enhancements or stability of APIs
+ * Doesn’t do much in terms of complex testing
+ * Needs some kind of regex capability for a more robust YAML testing capability
+ * Will require integration work to include it into the scorecard code.
+ * KUDO doesn’t currently make it easy to get the test Pod log output, this is useful to determine why a test might have failed, without it being built-into Kudo, the scorecard would need to read the Pod logs to present back to the end user.
```

Comment:
Is Kutll able to work with third-schemas? For example the OCP API?

Reply:
good point, I only tested this on kube and not on OCP, and the operator I tested (rqlite-operator) doesn't make use of any OCP-specific resource types.  I would need to confirm that there are no issues with OCP resources.